src/pages/patientView/timeline2/VAFChart.tsx

```diff
@@ -0,0 +1,801 @@
+        }
+    }
+
+    private tooltipFunction(tooltipData: any) {
+        return mutationTooltip(
+            tooltipData.mutation,
+            tooltipData.tooltipOnPoint
+                ? {
+                      mutationStatus: tooltipData.datum.mutationStatus,
+                      sampleId: tooltipData.datum.sampleId,
+                      vaf: tooltipData.datum.vaf,
+                  }
+                : undefined
+        );
+    }
+
+    @autobind
+    private getTooltipComponent() {
+        let mutationTooltip = this.props.wrapperStore.tooltipModel;
+        if (
+            !mutationTooltip ||
+            mutationTooltip.mouseEvent == null ||
+            mutationTooltip.mutation == null ||
+            mutationTooltip.datum == null
+        ) {
+            return <span />;
+        } else {
+            let tooltipPlacement =
+                mutationTooltip.mouseEvent.clientY < 250 ? 'bottom' : 'top';
+            return (
+                <Portal isOpened={true} node={document.body}>
+                    <Popover
+                        className={classnames(
+                            'cbioportal-frontend',
+                            'cbioTooltip',
+                            survivalStyles.Tooltip,
+                            styles.Tooltip
+                        )}
+                        positionLeft={mutationTooltip.mouseEvent.pageX}
+                        positionTop={
+                            mutationTooltip.mouseEvent.pageY +
+                            (tooltipPlacement === 'top' ? -7 : 7)
+                        }
+                        style={{
+                            transform:
+                                tooltipPlacement === 'top'
+                                    ? 'translate(-50%,-100%)'
+                                    : 'translate(-50%,0%)',
+                            maxWidth: 400,
+                        }}
+                        placement={tooltipPlacement}
+                    >
+                        {this.tooltipFunction(mutationTooltip)}
+                    </Popover>
+                </Portal>
+            );
+        }
+    }
+
+    @action
+    // Update store with groupBy sample tracks
+    setGroupByTracks(sampleGroups: { [index: number]: string[] }) {
+        const tracks: CustomTrackSpecification[] = [];
+        if (sampleGroups[-1] === undefined)
+            _.forIn(this.sampleGroups, (sampleIds: string[], key: string) => {
+                const index = parseInt(key);
+                tracks.push({
+                    renderHeader: () => this.groupByTrackLabel(index),
+                    renderTrack: () => this.sampleIconsGroupByTrack(sampleIds),
+                    height: () => this.groupIndexToTrackHeight[index],
+                    labelForExport: this.clinicalValuesForGrouping[index],
+                });
+            });
+        this.props.wrapperStore.groupByTracks = tracks;
+    }
+
+    yAxisHeaderReaction = autorun(() => {
```

Comment:
Please review the use of an autorun here.

Reply:
Hi Pim.  So, here's my analysis of the situation.   The problem is that the track header is a computation of this.ticks, but it doesn't have access to it because it's a property of a sibling component.  (This is just an unfortunate consequence of the fact that though header is conceptually a child of track, it cannot be rendered as part of the track.)   So, these autoruns observe local state and update the wrapperStore, which is effectively a parent of both track header and this component.   But I believe the right pattern would be just to bump ticks itself into the store. That way both header and chart can observe it properly and we don't need any of this autorun shenanigans.   In general, we should always try to avoid reactions (autorun) and rely instead on one way flow of data from parent to child.